### PR TITLE
Adapt the JSON Schema of the federation result message

### PR DESCRIPTION
**Pull request by 1florentin**
As discussed in our last meeting. This PR adds the challenge and public_key objects to the schema (+ validation in the form of oneOf)

### CODE REVIEW

**Review by onsriahi14 — APPROVED** [review]

File: protocol/query/method/message/data/dataFederationResult.json [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                "object": {
+                    "const": "federation"
+                },
+                "action": {
+                    "const": "result"
+                },
+                "status": {
+                    "type": "string",
+                    "$comment": "status of the authentication attempt"
+                },
```
Comment (arnauds5) [inline protocol/query/method/message/data/dataFederationResult.json:19]: The status value could be restricted to be some specific value ("failed", etc ?), either with pattern or with enum.
Or with const if only one value possible.

File: protocol/query/method/message/data/dataFederationResult.json [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                    "$ref": "../message.json",
+                    "$comment": "message/message containing a FederationChallenge data"
+                }
+            },
+            "additionalProperties": false,
+            "required": [
+                "object",
+                "action",
+                "status",
+                "challenge"
```
Comment (arnauds5) [inline protocol/query/method/message/data/dataFederationResult.json:34]: The reason field is missing in the required fields

File: protocol/query/method/message/data/dataFederationResult.json [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+                "object": {
+                    "const": "federation"
+                },
+                "action": {
+                    "const": "result"
+                },
+                "status": {
+                    "type": "string",
+                    "$comment": "status of the authentication attempt"
+                },
```
Comment (arnauds5) [inline protocol/query/method/message/data/dataFederationResult.json:49]: The status value could be restricted to be only the specific value "success", either with pattern or with enum or with const.

File: protocol/examples/messageData/federation_result/federation_result.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+    "action": "result",
+    "status": "success",
+    "challenge": {
+        "data": "eyJvYmplY3QiOiJmZWRlcmF0aW9uIiwiYWN0aW9uIjoiY2hhbGxlbmdlIiwidmFsdWUiOiJlYmEzZTI0ZWZjZDBiNTNmYTY5OTA4YmFkNWQxY2I2OTlkNzk4MGQ5MzEwOWRhMGIyYmZkNTAzN2MyYzg5ZWUwIiwidGltZXN0YW1wIjoxNzEzMzg1NTY4fQ==",
+        "sender": "zXgzQaa_NpUe-v0Zk_4q8k184ohQ5nTQhBDKgncHzq4=",
+        "signature": "BILYwYkT5tOBL4rCD7yvhBkhAYqRXOI3ajQ2uJ1gAk-g6nRc38vMMnlHShuNCQ3dQFXYZPn37cCFelhWGjY8Bg==",
+        "message_id": "sD_PdryBuOr14_65h8L-e1lzdQpDWxUAngtu1uwqgEI=",
+        "witness_signatures": []
+    }
+}
```
Comment (arnauds5) [inline protocol/examples/messageData/federation_result/federation_result.json:12]: This example doesn't seems to match a full success result message. There is no public_key field.
Maybe we should create two correct example, one for success and one for failure. And add this to the filename.

**Review by arnauds5 — CHANGES_REQUESTED** [review]

File: protocol/query/method/message/data/dataFederationResult.json (hunk 1 above)
Comment (1florentin) [reply protocol/query/method/message/data/dataFederationResult.json]: should be resolved in https://github.com/dedis/popstellar/pull/1832/commits/0929ebdb329ed2c75639139d90b9bfb5da64f0b0

File: protocol/query/method/message/data/dataFederationResult.json (hunk 2 above)
Comment (1florentin) [reply protocol/query/method/message/data/dataFederationResult.json]: should be resolved in https://github.com/dedis/popstellar/pull/1832/commits/0929ebdb329ed2c75639139d90b9bfb5da64f0b0

File: protocol/examples/messageData/federation_result/federation_result.json (hunk 4 above)
Comment (1florentin) [reply protocol/examples/messageData/federation_result/federation_result.json]: i added one correct positive case now, but you're right there should probably also be one for failure

**Review by arnauds5 — APPROVED** [review]

**Pull request merged**